src/amplitude/worker.py

```diff
@@ -39,7 +39,7 @@ def stop(self):
     def flush(self):
         events = self.storage.pull_all()
         if events:
-            self.threads_pool.submit(self.send, events)
+            self.send(events)
```

Comment:
@bohan-amplitude are we ok with this?

Reply:
The difference is flush event using main thread or threadpool. I was changing this because I don't want to block main thread. Right now many unittests need to wait flush finish then verify result. I talked with Xinyi and change back to send won't cause problem for most of the time. I think here it's better to let flush() return future object returned by threads_pool.submit(). In test, use the future object to wait for result before verify. In other use cases returned future can be ignored. @Mercy811 Can you take a look at [this](https://docs.python.org/3/library/concurrent.futures.html#future-objects) and try?